code/Git.md

```diff
@@ -0,0 +1,84 @@
+# Commit
+
+This gives a rough guide how to make good commits that are easy to review and understand.
+
+Most of these shouldn't be a blocker for getting a commit merged, however it's good practice. If you have difficulty in any of these parts, then ask for help.
+
+**NB: Github pull requests require the message and subject to be manually edited before merging.**
+
+## Subject
+
+Our commit messages follow quite closely [Go Project](https://github.com/golang/go/wiki/CommitMessage) conventions. See also https://golang.org/doc/contribute#commit_messages.
+
+```
+[scope]: [change]
+
+[longer description]
+```
+
+The `[scope]` part gives a quick insight into which packages it affects. This usually is just the package name. However, sometimes using `mod` for module changes or `ci` for Jenkins changes are applicable.
+
+If `[scope]` becomes long then using "bash expansion" syntax can be used to shorten it, e.g. `satellite/{metabase,metainfo}`. However, if that is still long, then it can be an indication that the commit should be split into multiple commits.
+
+The `[scope]` should avoid including the repository name, since when looking at the commit log, you already know the repository context.
+
+On rare cases, for fixing lint changes or refactoring, using scope `all` is useful. However, such changes should try to avoid making functional changes in the same commit.
+
+When a change requires minor change in another package, e.g. a method name in package `metabase` requires updating `metainfo` package, then it's not necessary to include the other package. Usually that doesn't add useful information.
+
+`[change]` should complete the blank in `This change modifies [scope] to ______`. It should start with lowercase and there's no trailing period.
+
+Try to keep the subject line less than \~76 characters, however, it's not critical. Subject line should always be followed by an empty line.
+
+---
+
+The first line helps to understand the impact of the change. For example when trying to find a possible cause with broken zombie deletion, then searching from this list is really effective:
+
+```
+$ git log --oneline
+5269596c7 satellite/{metabase,metainfo}: use ObjectStream as argument
+d06206488 satellite/rolluparchive: archiveAge use 90d prod default for storj-sim
+dba932148 satellite/metabase: Remove pending_index
+834b7cf6b private/server: close connection after each connector test
+efcb8e412 .github: enable dependabot for upgrading go dependencies
+e8ef68992 satellite/metabase: add DeleteZombieObjects method
+bb343d902 satellite/satellitedb: don't remove offline nodes from containment
+```
+
+These subject lines are also useful for creating change logs in https://github.com/storj/storj/releases and help undertand which commits can be omitted from the release.
+
+## Body
+
+Wrap the body at \~76 characters, to please git viewing tools, unless tables require otherwise.
```

Comment:
BTW Gerrit @ https://review.dev.storj.io/ wraps body at 72 characters. If someone wraps manually at 76 and Gerrit soft-wraps (if a line is "too long") at 72, the body might look a little bit funny.

Reply:
But that seems too difficult, so let's change it to 72.